src/pip/_internal/operations/install/wheel.py

```diff
@@ -484,6 +481,10 @@ def is_entrypoint_wrapper(name):
         # Ignore setuptools-generated scripts
         return (matchname in console or matchname in gui)
 
+    # Zip file path separators must be /
+    subdirs = list(set(p.split("/")[0] for p in wheel_zip.namelist()))
```

Comment:
Do we need to convert this back to `list`? Also there’s a minor optimisation available: `p.split("/", 1)`

Reply:
We don't, and good suggestion! I updated this PR and since this was lifted from `utils.wheel.wheel_dist_info_dir`, I've submitted #8529 to fix it there.